libnmstate/nm/sriov.py

```diff
@@ -67,13 +68,15 @@ def get_info(device):
     if not connection.profile:
         return info
 
-    sriov_setting = connection.profile.get_setting_by_name(
-        nmclient.NM.SETTING_SRIOV_SETTING_NAME
-    )
-
-    if sriov_setting:
-        info[Ethernet.SRIOV_SUBTREE] = {
-            Ethernet.SRIOV.TOTAL_VFS: sriov_setting.props.total_vfs
-        }
+    ifname = device.get_iface()
+    try:
+        with open(f"/sys/class/net/{ifname}/device/sriov_numvfs") as f:
+            info[Ethernet.SRIOV_SUBTREE] = {
+                Ethernet.SRIOV.TOTAL_VFS: int(f.read())
+            }
+    except FileNotFoundError:
+        raise NmstateInternalError(
+            "Could not read from /sys/class/net/{ifname}/device/sriov_numvfs"
```

Comment:
This is missing the `f` specifier for the format string. This mistake indicates that there should be a test to check the error condition.
Also, please compute the path for the path only once and then use it both in the `open()` command and the error message to ensure that they are always the same.

Reply:
Oops. I will add the test for this and build the path once.

Reply:
Fixed the `f ` specifier. About the test, please see comments below